zkwrapper/zk_queue.cc

```diff
@@ -0,0 +1,62 @@
+//
+// Created by Prudhvi Boyapalli on 10/17/16.
+//
+
+#include "zk_queue.h"
+#include <easylogging++.h>
+
+const std::string ZKQueue::CLASS_NAME = ": **ZkQueuer** : ";
+
+std::string q_path;
+std::string peek_path;
+std::string element;
+
+std::string ZKQueue::peek() {
+    return peek_path;
+}
+
+int ZKQueue::pop() {
+    int error_code;
+    if (zk.delete_node(peek_path, error_code)) {
+        LOG(ERROR) << CLASS_NAME <<  "The element '"<< peek_path << "' was popped.";
+        std::vector<std::string> children = std::vector<std::string>();
+        // TODO: Don't set children
+        if (zk.get_children(q_path, children, error_code)) {
+            std::sort(children.begin(), children.end());
+            peek_path = children.front();
+            LOG(ERROR) << CLASS_NAME <<  "The first element of the queue is now: " << peek_path;
+            return 0;
+        } else {
+            // TODO: Handle error better
```

Comment:
Yea, but this function returns a std::string...I think we should follow the same convention as in zkwrapper of passing in vars by ref to hold the return values. But, should making this change be in the scope of this PR?